File: week-04/day1/petrol-station.py
'''
Create Station and Car classes
Station
gas_amount
refill(car) -> decreases the gasAmount by the capacity of the car and increases the cars gas_amount
Car
gas_amount
capacity
create constructor for Car where:
initialize gas_amount -> 0
initialize capacity -> 100
'''

class Station():

    def __init__(self, gas_amount):
        self.gas_amount = gas_amount

    def refill(self, car):
        car.gas_amount += self.gas_amount
        self.gas_amount -= car.capacity

class Car():

    def __init__(self):
        self.gas_amount = 0
        self.capacity = 100


station = Station(2000)
car = Car()

station.refill(car)

print(car.gas_amount, station.gas_amount)

File: week-08/day2/petrol-station.js
'use strict';
// Create and object called car
//  - It should store its petrol level called petrolLevel
//  - It should store its petrol capacity called petrolCapacity
//  - It should have a refill(amount) method, that increments the petrol level,
//    than returns how much petrol it consumed from the given amount
//  - Initialize the petrol level to zero and the capacity to 50 
//
// Create an object called station
//  - It should store petrol amount called petrolStorage
//  - It should have a provide(car) method that calls the refill method of the car
//    with the stored petrol amount as a parameter, then decrement the used petrol
//  - Initialize the petrol amount to 3000

var car = {
    petrolLevel: 0,
    petrolCapacity: 50,
    refill: function(amount){
        var freeSpaceInTank = this.petrolCapacity - this.petrolLevel;
        if (amount < freeSpaceInTank){
            return amount;
        } else{
            return freeSpaceInTank;    
        }
    }
}
console.log(car.refill(45));


var station = {
    petrolStorage: 3000,
    provide: function(someCar, amount){
        var consumedPetrol = car.refill(amount);
        return this.petrolStorage - consumedPetrol;
    }
}

console.log(station.provide(car, 45));
console.log(car.petrolLevel);
console.log(station.petrolStorage);

